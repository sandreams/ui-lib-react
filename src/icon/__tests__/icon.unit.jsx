/**
 * @jest-environment jsdom
 */
import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../index';
import { render, fireEvent, screen } from '@testing-library/react';
describe('icon components', () => {
  it('test snapshots', () => {
    const json = renderer.create(<Icon name="editor" />).toJSON();
    expect(json).toMatchSnapshot();
  });
  it('test icon click', () => {
    const fn = jest.fn();
    const { queryByTestId, debug } = render(<Icon data-testid="aaa" name="editor" onClick={fn} />);
    // debug();
    fireEvent.click(queryByTestId('aaa'));
    expect(fn).toBeCalled();
  });
});
