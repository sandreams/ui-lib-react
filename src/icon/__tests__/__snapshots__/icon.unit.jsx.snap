// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon components test snapshots 1`] = `
<svg
  className="fui-icon"
>
  <use
    xlinkHref="#editor"
  />
</svg>
`;
